src/app-router.js

```diff
@@ -172,3 +150,53 @@ function targetIsThisWindow(target) {
     targetWindow === '_self' ||
     (targetWindow === 'top' && window === window.top);
 }
+
+function processResult(instruction, result, instructionCount, router) {
+  if (!(result && 'completed' in result && 'output' in result)) {
+    resut = result || {};
+    result.output = new Error(`Expected router pipeline to return a navigation result, but got [${JSON.stringify(result)}] instead.`);
```

Comment:
This threw an error that wasn't handled before. Now it's rolled up like any other navigation error.